src/ai/ai_config.cpp

```diff
@@ -128,13 +133,13 @@ void AIConfig::SetSetting(const char *name, int value)
 	ScriptConfig::SetSetting(name, value);
 }
 
-void AIConfig::AddRandomDeviation()
+void AIConfig::AddRandomDeviation(bool all)
 {
 	int start_date = this->GetSetting("start_date");
 
 	ScriptConfig::AddRandomDeviation();
 
 	/* start_date = 0 is a special case, where random deviation does not occur.
 	 * If start_date was not already 0, then a minimum value of 1 must apply. */
-	this->SetSetting("start_date", start_date != 0 ? max(1, this->GetSetting("start_date")) : 0);
+	this->SetSetting("start_date", all ? start_date != 0 ? max(1, this->GetSetting("start_date")) : 0 : start_date);
```

Comment:
Holy ternary batman, this is bad. At the *very least* this should have parentheses so the intended evaluation order of those nested ternaries is obvious, but really just don't introduce another ternary. Split it into a separate if statement.
This isn't smart or cool or fancy, it's just needlessly difficult to understand.